server.rb

```diff
@@ -0,0 +1,330 @@
+      logger.debug "#{@report}"
+      `rm -rf #{repository}`
+      @output = JSON.parse @report
+      annotations = []
+
+      # Rubocop reports the number of errors found in "offense_count"
+      if @output["summary"]["offense_count"] == 0
+        conclusion = 'success'
+      else
+        conclusion = 'neutral'
+        @output["files"].each do |file|
+
+          # Only parse offenses for files in this app's repository
+          file_path = file["path"].gsub(/#{repository}\//,'')
+          annotation_level = 'notice'
+
+          # Parse each offense to get details and location
+          file["offenses"].each do |offense|
+            start_line   = offense["location"]["start_line"]
+            end_line     = offense["location"]["last_line"]
+            start_column = offense["location"]["start_column"]
+            end_column   = offense["location"]["last_column"]
+            message      = offense["message"]
+
+            # Create a new annotation for each error
+            annotation = {
+              path: file_path,
+              start_line: start_line,
+              end_line: end_line,
+              start_column: start_column,
+              end_column: end_column,
+              annotation_level: annotation_level,
+              message: message
+            }
+            annotations.push(annotation)
+          end
+        end
+      end
+
+      # Updated check run summary and text parameters
+      summary = "Octo Rubocop summary\n-Offense count: #{@output["summary"]["offense_count"]}\n-File count: #{@output["summary"]["target_file_count"]}\n-Target file count: #{@output["summary"]["inspected_file_count"]}"
+      text = "Octo Rubocop version: #{@output["metadata"]["rubocop_version"]}"
+
+      # Mark the check run as complete! And if there are warnings, share them.
+      updated_check_run = @installation_client.patch(
+        "repos/#{@payload['repository']['full_name']}/check-runs/#{@payload['check_run']['id']}",
+        {
+          accept: 'application/vnd.github.antiope-preview+json',
+          name: 'Octo Rubocop',
+          status: 'completed',
+          conclusion: conclusion,
+          completed_at: Time.now.utc.iso8601,
+          output: {
+            title: "Octo Rubocop",
+            summary: summary,
+            text: text,
+            annotations: annotations
+          },
+          actions: [{
+            label: "Fix this",
+            description: "Automatically fix all linter notices.",
+            identifier: "fix_rubocop_notices"
+          }]
+        }
+      )
+
+    end
+
+    # Handles the check run `requested_action` event
+    # See https://developer.github.com/v3/activity/events/types/#checkrunevent
+    def take_requested_action
+      full_repo_name = @payload['repository']['full_name']
+      repository     = @payload['repository']['name']
+      head_branch    = @payload['check_run']['check_suite']['head_branch']
```

Comment:
This turns out to be a problem—When I open a PR, requesting to merge a branch `test` into `master`, like this one:
https://github.com/DEGoodmanWilson/internal-sandbox/pull/1/checks
The value of `head_branch` is `master`. So when the "fix changes" button gets clicked, the CI server is pushing the changes into the _wrong branch_ :sob: It's also checking out the wrong branch for running Rubocop on.

We want to check out, and push our changes onto the `test` branch, but I can't see where that information is available to us. All we have is the SHA1 for the commit that we are supposed to be checking, which is in `@payload['check_run']['check_suite']['head_sha']`. _this_ value is correct: On my PR, it resolves to `5530c5a92d3d4649329aa11cccff61aa26128569`, which is indeed the SHA for `$HEAD` on the `test` branch.

Reply:
Sorry about that! It might also be that the commit history is _seriously_ messed up. If it is working for you, then I am inclined to just say my situation is a fluke, and just ignore it.

Reply:
Oh, but I did add you as a collaborator :D